### PR TITLE
fixes for better errorhandling

### PR DESCRIPTION
**Pull request by ujann**
errormessage does not have to be a string, allow to return statuscode directly, also send headers (e.g. CORS!!) if Generic error

### CODE REVIEW

File: lib/proxyIntegration.js [hunk 1]
```diff
@@ -74,15 +79,20 @@ function normalizeRequestPath(event) {
...
-    }
-    return {statusCode: 500, body: `Generic error: ${JSON.stringify(error)}`};
+        if (typeof error.reason === 'string' && errorMapping && errorMapping[error.reason]) {
+            return {statusCode: errorMapping[error.reason], body: JSON.stringify(error.message), headers: headers};
+        }else if(typeof error.status === 'number'){
+            return {statusCode: error.status, body: JSON.stringify(error.message), headers: addCorsHeaders({})};
+        }
+        try{
+            return {statusCode: 500, body: `Generic error: ${JSON.stringify(error)}`, headers: addCorsHeaders({})};
+        }catch(stringifyError){}
```
Comment (chgohlke) [inline lib/proxyIntegration.js]: formatting code please

File: lib/proxyIntegration.js [hunk 2]
```diff
@@ -7,6 +7,13 @@ const NO_MATCHING_ACTION = request => {
...
 };
 
+function addCorsHeaders(toAdd) {
+    toAdd["Access-Control-Allow-Origin"] = "*";
+    toAdd["Access-Control-Allow-Methods"] = "GET,POST,PUT,DELETE,HEAD";
+    toAdd["Access-Control-Allow-Headers"] = "Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token";
+    return toAdd;
+}
+
 function process(proxyIntegrationConfig, event) {
```
Comment (chgohlke) [inline lib/proxyIntegration.js]: It is possible to cleanup the method 'process' a little bit?

**Comment by chgohlke** [comment]
Hi @ujann! There are some issues on Travic CI by tests. https://travis-ci.org/WeltN24/aws-lambda-router. Please, please fix this before we can make a release.

**Comment by ujann** [comment]
you mean the 2 stacktraces? these are ok, the framework logs sync exceptions to console...

File: lib/proxyIntegration.js (hunk 1 above)
Comment (ujann) [reply lib/proxyIntegration.js]: done

File: lib/proxyIntegration.js (hunk 2 above)
Comment (ujann) [reply lib/proxyIntegration.js]: true... well that would be a small refactoring session.... not in this PR

Comment (chgohlke) [reply lib/proxyIntegration.js]: ok, lets do next week or so 👍

**Pull request merged**